src/main/java/com/cairn/waypoint/dashboard/endpoints/protocol/GetAllProtocolsByUserIdEndpoint.java

```diff
@@ -45,9 +59,83 @@ public GetAllProtocolsByUserIdEndpoint(ProtocolDataService protocolDataService)
           @ApiResponse(responseCode = "403", description = "Forbidden",
               content = {@Content(schema = @Schema(hidden = true))})
       })
-  public ResponseEntity<List<Protocol>> getProtocolsByUserId(@PathVariable Long userId) {
+  public ResponseEntity<AccountProtocolListDto> getProtocolsByUserId(@PathVariable Long userId) {
     log.info("Fetching all protocols for user ID [{}]", userId);
-    List<Protocol> protocols = protocolDataService.findByUserId(userId);
-    return ResponseEntity.ok(protocols);
+    return ResponseEntity.ok(
+        AccountProtocolListDto.builder()
```

Comment:
This should become a mapper. This is a bit verbose and could be error prone if copied to other places